### PR TITLE
Code review Product CRUD

### PR DESCRIPTION
**Pull request by berki0**
I fixed the service and the controller of the USER model by adding validations when creating and updating, added tests, added a modelMapper

### CODE REVIEW

**Review by BelisAliosmanova — APPROVED** [review]

File: src/main/java/com/example/ludogoriesoft/lukeriaerpapi/models/User.java [hunk 1]
```diff
@@ -31,4 +31,6 @@ public class User {
     private String password;
     @Enumerated(EnumType.STRING)
     private Role role;
+    @Column(name = "is_deleted")
```
Comment (emiliyank) [inline src/main/java/com/example/ludogoriesoft/lukeriaerpapi/models/User.java:34]: The validations in the User model does not work. For example we can create a user with a username shorten than 4 symbols. 
Also the error messages does not align with the validations for password: 
`@NotNull(message = "Моля въведете парола с поне 5 символа")`

File: src/main/java/com/example/ludogoriesoft/lukeriaerpapi/services/CartonService.java [hunk 2]
```diff
@@ -20,10 +19,8 @@ public class CartonService {
     private final ModelMapper modelMapper;
 
     public List<CartonDTO> getAllCartons() {
-        List<Carton> cartons = cartonRepository.findByDeletedFalse();
-        return cartons.stream()
-                .map(carton -> modelMapper.map(carton, CartonDTO.class))
-                .toList();
+        List<Carton> laptops = cartonRepository.findByDeletedFalse();
```
Comment (emiliyank) [inline src/main/java/com/example/ludogoriesoft/lukeriaerpapi/services/CartonService.java:22]: laptops is not a proper name here

File: src/main/java/com/example/ludogoriesoft/lukeriaerpapi/controllers/UserController.java [hunk 3]
```diff
@@ -17,31 +18,37 @@
...
     @GetMapping("/{id}")
-    public ResponseEntity<UserDTO> getUserById(@PathVariable(name = "id") Long id) {
-        return ResponseEntity.ok(userService.getUserById(id));
+    public ResponseEntity<UserDTO> getUserById(@PathVariable(name = "id") Long id) throws ChangeSetPersister.NotFoundException {
+            UserDTO user = userService.getUserById(id);
+            return ResponseEntity.ok(user);
     }
 
     @PostMapping
     public ResponseEntity<UserDTO> createUser(@Valid @RequestBody User user) {
```
Comment (emiliyank) [inline src/main/java/com/example/ludogoriesoft/lukeriaerpapi/controllers/UserController.java:33]: Would be better to use `UserDTO` as a parameter instead of User. In this case the validations should be moved from the model to the `UserDTO`

File: src/main/java/com/example/ludogoriesoft/lukeriaerpapi/controllers/ProductController.java [hunk 4]
```diff
@@ -33,12 +34,12 @@ public ResponseEntity<ProductDTO> createPlate(@Valid @RequestBody ProductDTO pro
     }
 
     @PutMapping("/{id}")
-    public ResponseEntity<ProductDTO> updateProduct(@PathVariable("id") Long id, @Valid @RequestBody ProductDTO productDTO) {
+    public ResponseEntity<ProductDTO> updateProduct(@PathVariable("id") Long id, @Valid @RequestBody ProductDTO productDTO) throws ChangeSetPersister.NotFoundException {
```
Comment (emiliyank) [inline src/main/java/com/example/ludogoriesoft/lukeriaerpapi/controllers/ProductController.java:37]: Since we are working with the `ProductDTO` here I believe that the validation annotations like `@Min` should be added to the DTO and not to the model

File: src/test/java/com/example/ludogoriesoft/lukeriaerpapi/services/UserServiceTest.java [hunk 5]
```diff
@@ -2,296 +2,391 @@
...
+        // Act and Assert
+        assertThrows(jakarta.validation.ValidationException.class, () -> {
+            userService.updateUser(userId, userDTO);
+        });
+    }
+
 
-        when(userRepository.findById(userId)).thenReturn(Optional.empty());
+    @Test
+    void testCreateUser_UsernameBlank() {
```
Comment (emiliyank) [inline src/test/java/com/example/ludogoriesoft/lukeriaerpapi/services/UserServiceTest.java:326]: We can also have a test method for username with less that 4 symbols (the minimum required symbols) and it should not create the user.

**Review by emiliyank — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/ludogoriesoft/lukeriaerpapi/controllers/UserController.java (hunk 3 above)
Comment (berki0) [reply src/main/java/com/example/ludogoriesoft/lukeriaerpapi/controllers/UserController.java:33]: In order to avoid having a password field in UserDTO, we submit User user for saving. In the new commit, I tried to have validations in UserDTO

File: src/main/java/com/example/ludogoriesoft/lukeriaerpapi/controllers/ProductController.java (hunk 4 above)
Comment (berki0) [reply src/main/java/com/example/ludogoriesoft/lukeriaerpapi/controllers/ProductController.java:37]: i added validate in DTOs

File: src/test/java/com/example/ludogoriesoft/lukeriaerpapi/services/UserServiceTest.java (hunk 5 above)
Comment (berki0) [reply src/test/java/com/example/ludogoriesoft/lukeriaerpapi/services/UserServiceTest.java:326]: added try test for create with short name
